lib/src/main/scala/spinal/lib/ethernet/Utils.scala

```diff
@@ -0,0 +1,78 @@
+package spinal.lib.ethernet
+
+import spinal.core._
+
+import UserConfiguration._
+
+object rotateLeftByte {
```

Comment:
rotate operation could be added to the Bits data type directly :)
Maybe in a more general form ?
data: Bits, bias: UInt, sliceWidth : BitCount  ?

Reply:
Ummm...

Actually, there is `rotateLeft()` in BitVector,
```Scala
def rotateLeft(that: UInt): T = {
    ...
}
```

I wonder if it would be better to use `switch case` instead of `bits shift` one by one.

Reply:
> I wonder if it would be better to use switch instead of bits shift one by one.

Hmm, the main challenge is to ensure the synthesis tool can infer a proper barrel shifter using log2up(width) stages, that's why it is done in shifts sequence.

I would say the main limitation of the current implementation is that there is now parameter to specify how many bits to rotate per unit.